ethreceipts/ethreceipts_test.go

```diff
@@ -0,0 +1,357 @@
+	testChain, err = testutil.NewTestChain()
+	if err != nil {
+		log.Fatal(fmt.Errorf("NewTestChain failed: %w", err))
+	}
+
+	wallet, err := testChain.Wallet()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	toAddress := testutil.DummyAddr()
+	signed, err := wallet.NewTransaction(context.Background(), &ethtxn.TransactionRequest{
+		From: wallet.Address(),
+		To:   &toAddress,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	assert.NoError(t, err)
+
+	txn, _, err := wallet.SendTransaction(context.Background(), signed)
+	if err != nil {
+		log.Fatal(err)
+	}
+	assert.NoError(t, err)
+
+	receipt, err := receipts.GetTransactionReceipt(context.Background(), txn.Hash())
+	assert.NoError(t, err)
+
+	assert.True(t, receipt.Status == types.ReceiptStatusSuccessful, "Transaction couldn't found or failed.")
+}
+
+func TestReceiptsFinality(t *testing.T) {
+	provider, err := ethrpc.NewProvider("http://localhost:8545")
+	assert.NoError(t, err)
+
+	// Initializing Monitor
+	monitorOptions := ethmonitor.DefaultOptions
+
+	monitor, err := ethmonitor.NewMonitor(provider, monitorOptions)
+	assert.NoError(t, err)
+
+	go func(t *testing.T) {
+		err := monitor.Run(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}(t)
+	defer monitor.Stop()
+
+	logger := httplog.NewLogger("ethReceipts_test")
+
+	receipts, err := ethreceipts.NewReceipts(logger, provider, monitor, ethreceipts.Options{
+		NumBlocksUntilTxnFinality:     int(8),
+		WaitNumBlocksBeforeExhaustion: int(3),
+	})
+	assert.NoError(t, err)
+
+	go func(t *testing.T) {
+		err := receipts.Run(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}(t)
+	defer receipts.Stop()
+
+	// Ensure test-chain is running
+	testChain, err = testutil.NewTestChain()
+	if err != nil {
+		log.Fatal(fmt.Errorf("NewTestChain failed: %w", err))
+	}
+
+	wallet, err := testChain.Wallet()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	toAddress := testutil.DummyAddr()
+	signed, err := wallet.NewTransaction(context.Background(), &ethtxn.TransactionRequest{
+		From: wallet.Address(),
+		To:   &toAddress,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	assert.NoError(t, err)
+
+	txn, _, err := wallet.SendTransaction(context.Background(), signed)
+	if err != nil {
+		log.Fatal(err)
+	}
+	assert.NoError(t, err)
+
+	receipt, event, err := receipts.GetFinalTransactionReceipt(context.Background(), txn.Hash())
+	assert.NoError(t, err)
+	assert.True(t, event == ethreceipts.Finalized, "Transaction couldn't found or failed.")
+
+	currentBlockNumber, err := provider.BlockNumber(context.Background())
+	assert.NoError(t, err)
+
+	assert.True(t, currentBlockNumber-receipt.BlockNumber.Uint64() >= uint64(receipts.Options().NumBlocksUntilTxnFinality), "Transaction not finalized.")
+	assert.True(t, receipt.Status == types.ReceiptStatusSuccessful, "Transaction couldn't found or failed.")
+
+}
+
+// // // TestReceiptsWithReorg tests the receipt retrieval with a reorg. This is not a deterministic test, but it can be used to check
```

Comment:
whats with all these triple comments ..? and the code..?

is this PR ready for a merge.. or just a draft?